### PR TITLE
Jl/add unit testing

### PR DESCRIPTION
**Pull request by jordlay**
---
I've added UTs for the things that could be done easily without worrying about interacting with Azure.  That means the generate config and build commands.

They currently just check that the code doesn't hit any errors and that something is output.  We could later validate that the output is sensible, but that is a lot more effort and a job for another day.

In addition:
- I've tweaked the code so that file paths in the input files are relative to that input file (I think that is the right thing to do, but it also made writing UTs more intuitive).
- Fixed up some linting

Don't worry about reviewing the files in the mock_* folders, they are just example Helm charts and other artifacts.

### CODE REVIEW

File: src/aosm/azext_aosm/tests/latest/test_nsd.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import os
+from dataclasses import dataclass
+import json
+from unittest.mock import patch
+from tempfile import TemporaryDirectory
+from typing import Any, Dict
+
+from azext_aosm.custom import generate_design_config, build_design
+
+mock_nsd_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), "mock_nsd")
```
Comment (Cyclam) [inline src/aosm/azext_aosm/tests/latest/test_nsd.py:15]: Using pathlib could make this easier parse (per comment on `test_cnf.py`)

File: src/aosm/azext_aosm/tests/latest/test_nsd.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                self.network_function_definition_versions = NFDVs()
+
+        mock_client = AOSMClient()
+
+        class FakeCmd():
+            def __init__(self) -> None:
+                self.cli_ctx = None
+
+        cmd = FakeCmd()
+
```
Comment (Cyclam) [inline src/aosm/azext_aosm/tests/latest/test_nsd.py:84]: I suspect some of this set up could be useful to have in a fixture, but fine to leave that until we add another test that needs it.

**Review by Cyclam — COMMENTED** [review]
A few comments, but otherwise looks good. Thanks :-)

File: src/aosm/azext_aosm/_configuration.py [hunk 3]
```diff
@@ -117,7 +118,32 @@ class ArtifactConfig:
...
+        """
+        # If no path has been supplied we shouldn't try to update it.
+        if path == "":
+            return ""
+
+        # If it is an absolute path then we don't need to monkey around with it.
+        if os.path.isabs(path):
+            return path
+
+        return os.path.join(os.path.dirname(self.config_file), path)
```
[the remark this answers is not in the document]
Comment (sunnycarter) [reply src/aosm/azext_aosm/_configuration.py:142]: Is this going to change the behaviour of the quickstarts? Do the config files need changing?

Then - do we need something to know which version of CLI the quickstart works with? We don't and I imagine that's going to get gnarly (it already is)

[the remark this answers is not in the document]
Comment (jamiedparsons) [reply src/aosm/azext_aosm/_configuration.py:142]: I'm leaving this for now.  I don't think that `path.resolve()` is correct as it is the relative path from the config file that is supplied, not the relative path from the current location.

Empty strings mean that no value has been supplied and is handled as a special case above.

File: src/aosm/azext_aosm/tests/latest/test_nsd.py (hunk 2 above)
Comment (jamiedparsons) [reply src/aosm/azext_aosm/tests/latest/test_nsd.py:77]: I'm leaving this for now.

**Pull request merged**